File: src/main/java-reference/de/visustruct/view/StruktogrammPalette.java
package de.visustruct.view;

/**
 * Reihenfolge und feste Beschriftungen der linken Palette (englisch).
 */
public final class StruktogrammPalette {

	/** Indizes entsprechen {@link de.visustruct.control.Struktogramm}-Konstanten. */
	public static final int[] TYPEN_REIHENFOLGE = { 0, 1, 4, 5, 3, 2, 6, 7, 8 };

	/**
	 * Kurze englische Paletten-Texte; ausführliche Benennung steht im Tooltip.
	 */
	public static String getPaletteButtonLabel(int typ) {
		return getPaletteButtonKurzEnglish(typ);
	}

	/** Kompakte englische Beschriftung der Palette. */
	public static String getPaletteButtonKurzEnglish(int typ) {
		switch (typ) {
		case 0:
			return "Statement";
		case 1:
			return "If";
		case 2:
			return "Switch";
		case 3:
			return "For loop";
		case 4:
			return "While loop";
		case 5:
			return "Do-while loop";
		case 6:
			return "Endless loop";
		case 7:
			return "Break";
		case 8:
			return "Call";
		default:
			return "";
		}
	}

	/** Ausführliche englische Palette-Tooltips. */
	public static String getPaletteElementTooltipEnglish(int typ) {
		switch (typ) {
		case 0:
			return "Statement — simple action or assignment";
		case 1:
			return "If / else — decision";
		case 2:
			return "Switch — multi-way selection";
		case 3:
			return "For — counter-controlled loop";
		case 4:
			return "While — pre-test loop";
		case 5:
			return "Do-While — post-test loop";
		case 6:
			return "Infinite loop";
		case 7:
			return "Break — exit from a loop or switch";
		case 8:
			return "Call — procedure or method call";
		case 9:
			return "Empty placeholder block";
		default:
			return "";
		}
	}

	/** Bezeichnung für Leer-Element (Palette hat kein Kachel-Typ 9). */
	public static String getPaletteLabelLeerElement() {
		return "Empty";
	}

	/**
	 * Standardtext für neu eingefügte Blöcke (englisch / code-nah).
	 */
	public static String getDefaultTextForNewElement(int typ) {
		switch (typ) {
		case 0:
			return "Statement";
		case 1:
			return "condition";
		case 2:
			return "selector";
		case 3:
			return "i = 0; i < n; i++";
		case 4:
			return "condition";
		case 5:
			return "condition";
		case 6:
			return "\u221e";
		case 7:
			return "break";
		case 8:
			return "method()";
		case 9:
			return "\u00f8";
		default:
			return "";
		}
	}

	private StruktogrammPalette() {
	}
}

File: src/main/kotlin/de/visustruct/view/StruktogrammPalette.kt
package de.visustruct.view

/** Reihenfolge und feste Beschriftungen der linken Palette (englisch). */
object StruktogrammPalette {

    /** Indizes entsprechen [de.visustruct.control.Struktogramm]-Konstanten. */
    @JvmField
    val TYPEN_REIHENFOLGE = intArrayOf(0, 1, 4, 5, 3, 2, 6, 7, 8)

    @JvmStatic
    fun getPaletteButtonLabel(typ: Int): String = getPaletteButtonKurzEnglish(typ)

    @JvmStatic
    fun getPaletteButtonKurzEnglish(typ: Int): String =
        when (typ) {
            0 -> "Statement"
            1 -> "If"
            2 -> "Switch"
            3 -> "For loop"
            4 -> "While loop"
            5 -> "Do-while loop"
            6 -> "Endless loop"
            7 -> "Break"
            8 -> "Call"
            else -> ""
        }

    @JvmStatic
    fun getPaletteElementTooltipEnglish(typ: Int): String =
        when (typ) {
            0 -> "Statement — simple action or assignment"
            1 -> "If / else — decision"
            2 -> "Switch — multi-way selection"
            3 -> "For — counter-controlled loop"
            4 -> "While — pre-test loop"
            5 -> "Do-While — post-test loop"
            6 -> "Infinite loop"
            7 -> "Break — exit from a loop or switch"
            8 -> "Call — procedure or method call"
            9 -> "Empty placeholder block"
            else -> ""
        }

    @JvmStatic
    fun getPaletteLabelLeerElement(): String = "Empty"

    @JvmStatic
    fun getDefaultTextForNewElement(typ: Int): String =
        when (typ) {
            0 -> "Statement"
            1 -> "condition"
            2 -> "selector"
            3 -> "i = 0; i < n; i++"
            4 -> "condition"
            5 -> "condition"
            6 -> "\u221e"
            7 -> "break"
            8 -> "method()"
            9 -> "\u00f8"
            else -> ""
        }
}
